### PR TITLE
add default None -> UserWarning

### LINKED ISSUE #4746: MAINT, TST: no_deprecated_call modernization
As detailed at https://github.com/MDAnalysis/mdanalysis/pull/4744#discussion_r1807582150, `no_deprecated_call` (an old MDA testing utility) now requires an explicit warning class specification to work--so it'll choke with `warnings.warn("boo")`, but succeed with `warnings.warn("boo", UserWarning)`.

Should probably be fixed (and or/upstreamed to `pytest` or put in some `pytest` plugin perhaps).

It is still actively used in our suite, but only a little bit, and perhaps not enough to have tracked the increased complexity of the warnings system from Python 2->3.

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #4746

Changes made in this Pull Request:
 - This mimics the Python warnings library by setting `category=None` to `UserWarning` by default

We don't appear to test our utilities anywhere so I haven't added a formal test -- but running it with #4744 worked.

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4747.org.readthedocs.build/en/4747/

### CODE REVIEW

File: testsuite/MDAnalysisTests/util.py
Comment (tylerjereddy) [inline testsuite/MDAnalysisTests/util.py:240]: Ah, this is a surprisingly simple fix, very nice. Confirming it also works in my hands.

I'll approve but not merge yet given the release cycle situation.

**Review by tylerjereddy — APPROVED** [review]

**Review by IAlibay — CHANGES_REQUESTED** [review]
It would be best if this was applied to something in our tests - i.e. some kind of testing by application.

**Comment by orbeckst** [comment]
@tylerjereddy can you keep an eye on this PR, please? If you don't have the bandwidth, please unassign yourself again. Thanks.

**Comment by tylerjereddy** [comment]
@orbeckst @lilyminium testing locally, we could just follow through with our original intention here and the `develop` branch will fail this test without the patch that Lily is providing here. This fails on `develop` and passes in this branch:

```diff
--- a/testsuite/MDAnalysisTests/topology/test_itp.py
+++ b/testsuite/MDAnalysisTests/topology/test_itp.py
@@ -26,6 +26,7 @@ import MDAnalysis as mda
 import numpy as np
 from numpy.testing import assert_allclose, assert_equal
 
+from MDAnalysisTests.util import no_deprecated_call
 from MDAnalysisTests.topology.base import ParserBase
 from MDAnalysisTests.datafiles import (
     ITP,  # GROMACS itp
@@ -490,3 +491,5 @@ def test_missing_elements_no_attribute():
         u = mda.Universe(ITP_atomtypes)
     with pytest.raises(AttributeError):
         _ = u.atoms.elements
+    with no_deprecated_call():
+        mda.Universe(ITP_atomtypes)
```

That's what we meant to do all along anyway, and should match Irfan's request as well--actually using the functionality to test it.

**Comment by orbeckst** [comment]
Sounds good @tylerjereddy !

**Comment by tylerjereddy** [comment]
I rebased and pushed in my suggested adjustment.

**Review by IAlibay — APPROVED** [review]
Thanks, lgtm!

**Comment by IAlibay** [comment]
Looks like failures are related to the rdkit bug, so this should still be good to go.

**Comment by RMeli** [comment]
@IAlibay are you suggesting to merge despite the failures?

**Comment by IAlibay** [comment]
> @IAlibay are you suggesting to merge despite the failures?

If folks don't mind then yes, we'll need to fix the rdkit failure at some point but that's probably not happening today.

**Comment by tylerjereddy** [comment]
Ok, I squash-merged given the exact same CI failures are seen in other recent unrelated PRs.

Technically I pushed in a commit here so that's a bit like self-merging, but obviously the changes are very small, not really user-facing, and the approval was from another core dev.

**Pull request merged**
